### ISSUE TITLE
Rest API: some filters in dcim_devices_list changed unexpectedly in new nautobot release (from list of Names or IDs to single ID)

### ISSUE THREAD

**Issue by gioccher** [nautobot/nautobot#6656]
### Environment
* Nautobot version (Docker tag too if applicable): 2.3.13
* Python version: 3.9 and what's running on demo.nautobot.com
* Database platform, version: Mysql and what's running on demo.nautobot.com
* Middleware(s):

### Steps to Reproduce
GET https://demo.nautobot.com/api/dcim/devices/?role__n=leaf&role__n=spine
or look at documentation https://demo.nautobot.com/api/docs/#/dcim/dcim_devices_list

### Expected Behavior
In nautobot 2.2.9 it used to be possible to specify multiple values for role__n, status__n, contacts__n. Those values could either be the names or the IDs of the roles, statuses, or teams.

### Observed Behavior
In nautobot 2.3.13 it's only possible to specify a single ID.
Some filters like rack__n still allow multiple names or IDs, but the filters mentioned before now only allow a single ID.
The auto-generated documentation still mentions that both name or ID are accepted (like "Exclude Role (name or ID)")

**Comment by timizuoebideri1** [comment]
This stopped working after this modification to `RoleModelFilterSetMixin` https://github.com/nautobot/nautobot/pull/5668/files#diff-546ee051cefe9f31a4f9c85df5c069db0f36a0318213fa124517e94cc3658e4bR306-R316

**Comment by glennmatthews** [comment]
Good catch. Probably an order-of-operations issue, guessing status and role filters are being added *after* the automatic generation of the lookup-expression variants?

**Comment by glennmatthews** [comment]
Potentially related: #6456
